exasol/bucketfs/saas_file_api.py

```diff
@@ -0,0 +1,116 @@
+from __future__ import annotations
+from typing import Protocol, Literal, Iterable, ByteString, BinaryIO
+from datetime import datetime
+
+
+class Unset:
+    """
+    This is a temporary class that will be replaced with relevant construct from
+    SaaS Open API, once it is available.
+
+    """
+    def __bool__(self) -> Literal[False]:
+        return False
+
+
+UNSET: Unset = Unset()
+
+SAAS_FOLDER = 'folder'
+
+
+class SaasFile:
+    """
+    This is a temporary class that will be replaced with the File definition in the
+    SaaS Open API, once it is available.
+    """
+    name: str
+    type: str
+    path: str
+    last_modified: datetime
+    size: Unset | int = UNSET
+    children: Unset | list["SaasFile"] = UNSET
+
+
+class SaasFileApi(Protocol):
```

Comment:
if this is contained in a `saas` package/module naming can be simplified:

`File`
`FileApi`
...